core/src/Kokkos_Complex.hpp

```diff
@@ -44,6 +44,11 @@ class
     alignas(2 * sizeof(RealType))
 #endif
         complex {
+  static_assert(std::is_floating_point_v<RealType> &&
```

Comment:
Shall we merge this and then add a follow on PR with:

```suggestion
  static_assert((std::is_floating_point_v<RealType> &&
                        std::is_same_v<RealType, std::remove_cv_t<RealType>) ||
                        kokkos_type_is_half<RealType>::value)
```

Reply:
I do not see any usage of `Kokkos::complex` with `half_t` or `bhalf_t` in Trilinos:
https://github.com/search?q=repo%3Atrilinos%2FTrilinos+Kokkos%3A%3Acomplex+half_t&type=code

Reply:
Except that
* `kokkos_type_is_half` is not yet a thing
* that including `<Kokkos_Half.hpp>` is not acceptable

so we would have to explore some other mechanism to enable if we decide to add support.